azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/util/DeploymentTemplateBuilder.java

```diff
@@ -0,0 +1,356 @@
+import org.jclouds.azurecompute.arm.domain.VirtualNetwork.VirtualNetworkProperties;
+import org.jclouds.azurecompute.arm.domain.VirtualNetwork.AddressSpace;
+import org.jclouds.compute.domain.Template;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class DeploymentTemplateBuilder {
+
+   private String name;
+   private String group;
+   private Template template;
+   private AzureComputeArmTemplateOptions options;
+   private List<ResourceDefinition> resources;
+   private Map<String, String> variables;
+   private String loginUser;
+   private String loginPassword;
+   private String location;
+
+   public static final String DEFAULT_LOGIN_USER = "jclouds";
+   public static final String DEFAULT_LOGIN_PASSWORD = "Password1!";
+   private static final String DEPLOYMENT_MODE = "Incremental";
+
+   public DeploymentTemplateBuilder(String group, String name, Template template) {
+      this.name = name;
+      this.group = group;
+      this.template = template;
+      this.options = template.getOptions().as(AzureComputeArmTemplateOptions.class);
+      this.variables = new HashMap<String, String>();
+      this.resources = new ArrayList<ResourceDefinition>();
+      this.loginUser = options.getLoginUser() == null ? DEFAULT_LOGIN_USER : options.getLoginUser();
+      this.loginPassword = options.getLoginPassword() == null ? DEFAULT_LOGIN_PASSWORD : options.getLoginPassword();
+      this.location = template.getLocation().getId();
+   }
+
+   public String getLoginUserUsername() {
+      return loginUser;
+   }
+
+   public String getLoginPassword() {
+      return loginPassword;
+   }
+
+   public Template getTemplate() {
+      return template;
+   }
+
+   public DeploymentBody getDeploymentTemplate() {
+
+      addStorageResource();
+      addVirtualNetworkResource();
+      addPublicIpAddress();
+      addNetworkInterfaceCard();
+      addVirtualMachine();
+
+      DeploymentTemplate template = DeploymentTemplate.builder()
+              .$schema("https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#")
+              .contentVersion("1.0.0.0")
+              .resources(resources)
+              .variables(variables)
+              .parameters(DeploymentTemplate.Parameters.create())
+              .build();
+
+      DeploymentBody body = DeploymentBody.create(template, DEPLOYMENT_MODE, DeploymentTemplate.Parameters.create());
+
+      return body;
+   }
+
+
+   private void addStorageResource() {
+      String storageAccountName = name.replaceAll("[^A-Za-z0-9 ]", "") + "storage";
+
+      variables.put("storageAccountName", storageAccountName);
+
+      ResourceDefinition storageAccount = ResourceDefinition.builder()
+              .name("[variables('storageAccountName')]")
+              .type("Microsoft.Storage/storageAccounts")
+              .location(location)
+              .apiVersion("2015-06-15")
```

Comment:
Move this to a constant so the value is shared by the APIs too? (consider this elsewhere).